### ISSUE TITLE
New project crashes on Android with IllegalStateException

### ISSUE THREAD

**Issue by hippietilley** [facebook/react-native#24768]
## 🐛 Bug Report
Android app crashes after running `react-native run-android`. I have tried running the app on both a Pixel 3 xl and a Zebra MC40(running Android Lollipop). I did see a lot of issues here that seemed to be the same, but they had all been closed by the bot for not following the template.

## To Reproduce
- `react-native init AwesomeProject`
- connect an android device with usb debugging enabled
- `react-native run-android`

## Expected Behavior
I would expect the freshly initialized project to run on the device and display "Hello World!"

## Code Example
The code is exactly what you get when first running `react-native init [insert project name]`

The logcat output containing the stack trace for the exception: https://gist.github.com/hippietilley/dd5bf85e1dcb502b805569e4a003b184#file-adb_logcat-txt

## Environment
https://gist.github.com/hippietilley/dd5bf85e1dcb502b805569e4a003b184#file-react-native_info-txt

**Comment by hippietilley** [comment]
I'm not sure what else you need bot....

**Comment by matt-oakes** [comment]
@hippietilley 👋  Ignore the bot, the report looks fine.

I have run the commands you provided on my machine and can't reproduce the issue. A few questions:

* Are you making any other changes to the project files before you run it?
* Are you manually running the Metro packager (`yarn start`) or letting it open a new terminal window when you run `react-native run-android`?
* Does the Metro packager start with no errors?
* Can you try starting the Metro packager using `yarn start --reset-cache` and then run `react-native run-android` to see if that helps?

**Comment by hippietilley** [comment]
@matt-oakes, thanks for replying! When I first tried it, I did change the code (I added a button that did nothing). I tried it again without changing the code and got the same error both times. I have not started the Metro packager manually. It always opens a new terminal on it's own. I also haven't seen any errors from Metro.

After running `yarn start --reset-cache` and then running `react-native run-android` the app installed and ran. It did take 4m and 32s to build though, so for those of you reading this in the future because you also had this issue, go get some coffee and be patient.
